assets/training/model_management/src/run_model_preprocess.py

```diff
@@ -131,7 +133,18 @@ def run():
             preprocess_args.update(download_details.get("properties", {}))
             preprocess_args["misc"] = download_details.get("misc", [])
 
-    preprocess_args["task"] = task_name if task_name else preprocess_args.get("task")
+    if task_name is None or not SupportedTasks.has_value(task_name.lower()):
+        task_name = preprocess_args.get("task")
+        logger.warning("task_name is not provided or not supported. "
+                       f"Using task_name={task_name} from model download metadata.")
+
+    if task_name is None:
+        raise AzureMLException._with_error(
```

Comment:
This will throw exception with: 
`Unsupported task type provided. Supported task types are [......]`

Reply:
Yes